feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/stats.c

```diff
@@ -275,6 +275,21 @@ bool target_stats_survey_convert(radio_entry_t *radio_cfg, radio_scan_type_t sca
 {
 	target_survey_record_t delta;
 
+	if (data_new->info.chan != data_old->info.chan) {
+		/* Do not consider the old channel's data */
+		memset(data_old, 0, sizeof(*data_old));
```

Comment:
I think that is not true. In my debug logs, I saw data not reset to zero after a channel change. I will cross-check this again.

Reply:
So if we reset the data_old to 0, then the delta from data_new -> 0 could be very large and incorrect.
Perhaps better if we just report all the values (including the duration) as 0, so that this transitional sample has no impact on the measurement.  Then at the next sample we will have good values of data_new and data_old from the same channel.

Discussion thoughts only, I don't know if the surmised effect happens nor if the cloud side can handle a sample with all values including duration = 0.

Reply:
Very large or incorrect may be possible with this change.
If we report any data with duration=0, either the GW or the UI see that as a missing report. We see a gap in UI during that report.

We could remove this channel-change related new code and live with differentiating between the old channel's data and the new.